src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionsLoaderTest.java

```diff
@@ -387,4 +387,63 @@ public void testXpathNoCheckYesId() throws Exception {
             .hasSize(1);
     }
 
+    @Test
+    public void testLoadXpathConfiguration() throws Exception {
+        final String[] configFiles = {
+            "InputSuppressionsLoaderConfig4.xml",
+            "InputSuppressionsLoaderConfig5.xml",
+            "InputSuppressionsLoaderConfig9.xml",
+            "InputSuppressionsLoaderConfig10.xml",
+        };
+
+        for (String configFile : configFiles) {
+            final String config = getPath(configFile);
+            SuppressionsLoader.loadXpathSuppressions(config);
+            final Set<TreeWalkerFilter> filterSet =
+                    SuppressionsLoader.loadXpathSuppressions(config);
+
+            final Set<TreeWalkerFilter> expectedFilterSet = new HashSet<>();
+            final XpathFilterElement xPath =
+                new XpathFilterElement("file1", "test", null, "id1", "//CLASS_DEF");
+            expectedFilterSet.add(xPath);
+            assertWithMessage("")
+                    .that(expectedFilterSet)
+                    .isEqualTo(filterSet);
+
+        }
+    }
+
+    @Test
+    public void testLoadSuppressionFilterConfiguration() throws Exception {
+        final String[] configFiles = {
+            "InputSuppressionsLoaderConfig1.xml",
+            "InputSuppressionsLoaderConfig2.xml",
+            "InputSuppressionsLoaderConfig3.xml",
+            "InputSuppressionsLoaderConfig6.xml",
+            "InputSuppressionsLoaderConfig7.xml",
+            "InputSuppressionsLoaderConfig8.xml",
+        };
+
+        for (String configFile : configFiles) {
+            final FilterSet fc =
+                SuppressionsLoader.loadSuppressions(getPath(configFile));
+            final SuppressFilterElement suppressElement = (SuppressFilterElement) fc.getFilters()
+                    .toArray()[0];
+
+            final String filePattern = TestUtil.getInternalState(suppressElement, "filePattern");
```

Comment:
I know that we still have a lot of bad examples of pure unit testing.
But we agreed by all that we are not doing this anymore, as it is useless code generation that doesn't not give much in quality value.

Please use Inputs with some check and suppression of violation.

Reply:
Done